app/controllers/assets_controller.rb

```diff
@@ -207,6 +209,13 @@ def asset_params
     )
   end
 
+  def check_for_missing_attachment
+    unless @asset.audio_file.attached?
+      flash.now[:error] = "Uh oh, there's no longer an mp3 attached to this track.<br/>
+        Please edit it and click 'upload a new version' to add the mp3"
```

Comment:
It's really weird to use flash for this because you only really need the session when redirecting.

Reply:
Do you mean in general you don't like `flash.now`? We have a UI pattern around flash that makes it trivial to display messages consistently, however we are slowly moving to a different UI that is js based, since many of our pages are turbolinks/ajaxy.

The ideal UI I'd like for this particular case is to modify the playlist player and overlay a message for both users/owners, disabling the play button.

Reply:
Oops, I misread that as using the regular flash.

And yes, I don't like flash or the Android toaster because I end up missing important information. But maybe that's a personal preference.